python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,469 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import print_function, division
+
+import os.path
+import sys
+
+import numpy as np
+
+import lsst.daf.persistence as dafPersist
+import lsst.pipe.base as pipeBase
+from lsst.afw.table import SourceCatalog, SchemaMapper, Field
+from lsst.afw.table import MultiMatch, SimpleRecord, GroupView
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+# import lsst.afw.coord as afwCoord
+
+from .plotAstrometryPhotometry import plotAstrometry, plotPhotometry, plotPA1
+from .calcSrd import computeWidths, getRandomDiff, calcPA1, calcPA2
+
+def getCcdKeyName(dataid):
+    """Return the key in a dataId that's referring to the CCD or moral equivalent.
+
+    @param dataid  A dictionary that will be searched for a key that matches
+        an entry in the hardcoded list of possible names for the CCD field.
+
+    Different camera mappings use different keys
+      to indicate the different amps/ccds in the same exposure.
+    We here look through the reference dataId to determine which one this.
+    """
+    possibleCcdFieldNames = ['ccd', 'ccdnum', 'camcol']
+
+    for name in possibleCcdFieldNames:
+        if name in dataid:
+            return name
+    else:
+        return None
+
+
+def isExtended(source, extendedKey, extendedThreshold=1.0):
+    """Is the source extended attribute above the threshold.
+
+    Higher values of extendedness indicate a resolved object
+    that is larger than a point source.
+    """
+    return source.get(extendedKey) >= extendedThreshold
+
+def averageRaDec(cat):
+    """Calculate the RMS for RA, Dec for a set of observations an object."""
+    ra = np.mean(cat.get('coord_ra'))
+    dec = np.mean(cat.get('coord_dec'))
+    return ra, dec
+
+# Some thoughts from Paul Price on how to do the coordinate differences correctly:
```

Comment:
Because it's not necessary for the photometric repeatability.  The goal is to do this correctly as part of calculating the astrometric SRD quantities, which is DM-4956.